test/api_save_product_test.dart

```diff
@@ -505,13 +510,150 @@ Like that:
       expect(product.nutriments, isNull);
     });
 
-    test('Nutriments', () async {
+    test('With nutriments', () async {
       Product product = Product(
           nutriments: Nutriments.empty()
             ..setValue(Nutrient.salt, PerSize.oneHundredGrams, 1.0));
 
       expect(product.noNutritionData, isFalse);
       expect(product.nutriments, isNotNull);
     });
+
+    test('Nutriments modifiers (server call)', () async {
+      Product product = Product(
+        barcode: '1234567890123',
+        nutriments: Nutriments.empty()
+          ..setValue(
+            Nutrient.proteins,
+            PerSize.oneHundredGrams,
+            null,
+            modifier: NutrientModifier.notProvided,
+          )
+          ..setValue(
+            Nutrient.energyKJ,
+            PerSize.oneHundredGrams,
+            1.0,
+            modifier: NutrientModifier.lessThan,
+          )
+          ..setValue(
+            Nutrient.salt,
+            PerSize.oneHundredGrams,
+            1.0,
+            modifier: NutrientModifier.approximately,
+          )
+          ..setValue(
+            Nutrient.magnesium,
+            PerSize.oneHundredGrams,
+            1.0,
+            modifier: NutrientModifier.greaterThan,
+          )
+          ..setValue(Nutrient.fat, PerSize.oneHundredGrams, 1.0),
+      );
+
+      final Status status = await OpenFoodAPIClient.saveProduct(
+        TestConstants.TEST_USER,
+        product,
+        uriHelper: uriHelper,
+      );
+
+      expect(status.status, 1);
+      expect(status.error, null);
+
+      final ProductResultV3 result = await OpenFoodAPIClient.getProductV3(
+        ProductQueryConfiguration(
+          product.barcode!,
+          language: OpenFoodFactsLanguage.GERMAN,
+          fields: null,
+          version: ProductQueryVersion.v3,
+        ),
+        user: TestConstants.TEST_USER,
+        uriHelper: uriHelper,
+      );
+
+      expect(result.product!.nutriments, isNotNull);
+      expect(
+        result.product!.nutriments?.getModifier(
+          Nutrient.proteins,
+          PerSize.oneHundredGrams,
+        ),
+        NutrientModifier.notProvided,
```

Comment:
Rather keeps the nutrient x modifier in a map, then save the product, then get the product, then check that the product matches the map.